source/OpenCDMSession.cpp

```diff
@@ -264,6 +264,48 @@ void OpenCDMSession::addProtectionMeta(GstBuffer *buffer, GstBuffer *subSample,
     rialto_mse_add_protection_metadata(buffer, info);
 }
 
+bool OpenCDMSession::addProtectionMeta(GstBuffer *buffer, GstCaps *caps)
+{
+    if (!caps)
+    {
+        TRACE_L1("Invalid caps");
+        return false;
+    }
+
+    GstStructure *structure = gst_caps_get_structure(caps, 0);
+    if (!structure)
+    {
+        TRACE_L1("Cannot get the structure from the caps");
+        return false;
+    }
+
+    GstStructure *info = gst_structure_copy(structure);
+    gst_structure_set_name(info, "application/x-cenc");
+    gst_structure_set(info, "mks_id", G_TYPE_INT, mRialtoSessionId, NULL);
+
+    if (!gst_structure_has_field_typed(info, "encrypted", G_TYPE_BOOLEAN))
```

Comment:
Do we want to set these or leave it all down to the app?

Reply:
I'd say don't set cipher-mode. As we set the others previously maybe safer to follow that pattern for now?

Reply:
removed cipher-mode, but added iv_size as this is not set in Cobalt but required by rialto-gstreamer.